applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.cpp

```diff
@@ -0,0 +1,296 @@
+    }
+
+    for(auto & var_settings : Parameters["supersonic_boundary_conditions"])
+    {
+        ReadBoundaryCondition(mSupersonicBCs, var_settings);
+    }
+
+    KRATOS_CATCH("");
+}
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitialize()
+{
+    NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+}
+
+
+void ApplyMachDependentBoundaryConditions::ExecuteInitializeSolutionStep()
+{
+    if(mRefreshNormalsEveryTimeStep)
+    {
+        NormalCalculationUtils().CalculateUnitNormals<Condition>(*mpModelPart);
+    }
+
+    // Enabling and disabling BC according to the time and active interval
+    const double time = mpModelPart->GetProcessInfo().GetValue(TIME);
+
+    for(auto & boundary_condition: mSupersonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    for(auto & boundary_condition: mSubsonicBCs)
+    {
+        boundary_condition.ActivateIfInsideTimeInterval(time);
+    }
+
+    // Enforcing boundary conditions
+    block_for_each(mpModelPart->Nodes(), [&](NodeType & rNode)
+    {
+        // Computing mach projected onto normal
+        const auto & normal = rNode.FastGetSolutionStepValue(NORMAL);
+        const auto & flow_direction = rNode.FastGetSolutionStepValue(*mpFlowDirectionVariable);
+
+        const double & mach = rNode.GetValue(MACH);
+        const double mach_projection = mach * inner_prod(normal, flow_direction) / norm_2(flow_direction);
+        
+        // Chosing BC set to enforce according to mach
+        const bool supersonic = fabs(mach_projection) >= 1.0;
+        const auto & active_bc  = supersonic ? mSupersonicBCs : mSubsonicBCs;
+
+        for(const auto & boundary_condition: active_bc)
+        {
+            boundary_condition.Enforce(rNode);
+        }
+
+    });
+}
+
+void ApplyMachDependentBoundaryConditions::ExecuteFinalizeSolutionStep()
+{
+    block_for_each(mpModelPart->Nodes(), [&](NodeType & rNode)
+    {
+        for(const auto & boundary_condition: mSupersonicBCs)
+        {
+            boundary_condition.Free(rNode);
+        }
+
+        for(const auto & boundary_condition: mSubsonicBCs)
+        {
+            boundary_condition.Free(rNode);
+        }
+    });
+}
+
+
+ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::
+    BoundaryConditionUtility(const std::string & variable_name,
+                            const double Value,
+                            const IntervalUtility & rIntervalUtility)
+    : mValue(Value), mInterval(rIntervalUtility)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF_NOT(KratosComponents<Variable<double>>::Has(variable_name))
+        << "There is no double variable named " << variable_name << std::endl;
+
+    mpVariable = &KratosComponents<Variable<double>>::Get(variable_name);
+
+    KRATOS_CATCH("")
+}
+
+
+void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::
+    ActivateIfInsideTimeInterval(const double time)
```

Comment:
```suggestion
void ApplyMachDependentBoundaryConditions::BoundaryConditionUtility::ActivateIfInsideTimeInterval(const double Time)
```
Style guide (single argument functions in one line and CamelCase for arguments.